src/server/pps/server/api_server_test.go

```diff
@@ -508,6 +508,45 @@ func TestCreatePipelineMultipleNames(t *testing.T) {
 	require.NoError(t, protojson.Unmarshal([]byte(resp.EffectiveCreatePipelineRequestJson), &req), "unmarshalling effective JSON must not error")
 	require.Equal(t, int64(4), req.DatumTries, "default and spec names map")
 	require.False(t, req.Autoscaling, "spec must override default")
+	require.Equal(t, req.Salt, "mysalt", "default must apply if not overridden")
+	// validate that the user and effective specs are correct
+	r, err := env.PachClient.PpsAPIClient.InspectPipeline(ctx, &pps.InspectPipelineRequest{Pipeline: &pps.Pipeline{Name: pipeline}})
+	if err != nil {
```

Comment:
Why not use `require.NoError`?

Reply:
No particular reason here, where I call `t.Fatal`; I could, and honestly to be consistent with the rest of our tests I arguably should. I think I was just writing normal Go error-handling code (`err := … ; if err != nil { … }`) — `require.Whatever` not being idiomatic _is_ an argument against it, and the `Fatal`/`Error` distinction can be really useful. But I wasn’t really being that systematic, e.g. the check at [l. 524](https://github.com/pachyderm/pachyderm/pull/9167/files/a88105fa8f3ac414ff3fa5072c8dcccf3669aca9#diff-871a18b8106b8eb6a54b5a150080dd1786f670f1cae25ca2cbfe34a085fb3dd6R524) should probably be fatal. 

I’ll rewrite to use package `require`. IMHO we should stop using it, but that should be a deliberate decision as a team.